irohad/main/impl/consensus_init.cpp

```diff
@@ -74,20 +74,8 @@ namespace iroha {
       }
 
       auto YacInit::createTimer(std::chrono::milliseconds delay_milliseconds) {
-        return std::make_shared<TimerImpl>([delay_milliseconds, this] {
-          // static factory with a single thread
-          //
-          // observe_on_new_thread -- coordination which creates new thread with
-          // observe_on strategy -- all subsequent operations will be performed
-          // on this thread.
-          //
-          // scheduler owns a timeline that is exposed by the now() method.
-          // scheduler is also a factory for workers in that timeline.
-          //
-          // coordination is a factory for coordinators and has a scheduler.
-          return rxcpp::observable<>::timer(
-              std::chrono::milliseconds(delay_milliseconds), coordination_);
-        });
+        return std::make_shared<TimerImpl>(delay_milliseconds,
+                                           rxcpp::observe_on_new_thread());
```

Comment:
I think here we can consider an opportunity to reuse thread from MST-propagator. I am okay with both fix or todo.